astrovascpy/scipy_petsc_conversions.py

```diff
@@ -407,23 +394,14 @@ def PETScVec2array(x):
     istart, iend = x.getOwnershipRange()
 
     nloc = iend - istart
-    Istart = MPI_COMM.gather(istart, root=0)
-    Iend = MPI_COMM.gather(iend, root=0)
+    Nloc = tuple(MPI_COMM.allgather(nloc))
+    Istart = tuple(MPI_COMM.allgather(istart))
 
     if MPI_RANK == 0:
         v = np_zeros(n, PETSc.ScalarType)
     else:
         v = None
 
-    if MPI_RANK == 0:
-        v[:nloc] = vloc
-
-    for iproc in range(1, MPI_SIZE):
-        if MPI_RANK == 0:
-            i0 = Istart[iproc]
-            i1 = Iend[iproc]
-            MPI_COMM.Recv(v[i0:i1], source=iproc, tag=77)
-        elif MPI_RANK == iproc:
-            MPI_COMM.Send(vloc, dest=0, tag=77)
+    MPI_COMM.Gatherv(vloc, [v, Nloc, Istart, MPI.DOUBLE], root=0)  # MPIU_REAL
```

Comment:
here you can use the `from_numpy_dtype(PETSc.ScalarType)` as used here: https://github.com/BlueBrain/AstroVascPy/blob/b7e501f6ba9dfb674e12186cbc00576c69c68575/astrovascpy/scipy_petsc_conversions.py#L224